restapi/pkg/controllers/router.go

```diff
@@ -0,0 +1,221 @@
+package controllers
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/OlehEngineer/goTasks/goTasks/restapi/pkg/domian"
+	"github.com/OlehEngineer/goTasks/goTasks/restapi/pkg/service"
+)
+
+type httpHandler struct {
+	service service.Service
+	engine  *echo.Echo
+}
+
+func New(service service.Service) *httpHandler {
+	h := &httpHandler{
+		service: service,
+		engine:  echo.New(),
+	}
+	return h
+}
+func (self *httpHandler) Run() {
+	self.RegisterRouters()
+	self.engine.Logger.Fatal(self.engine.Start(":8080"))
+}
+
+func (self *httpHandler) RegisterRouters() {
+	self.engine.GET("/api/v1/users", self.getUsersPagination)
+	self.engine.GET("/api/v1/users/:id", self.getUser)
+	self.engine.DELETE("/api/v1/users/:id", self.deleteUser)
+	self.engine.POST("/api/v1/users", self.postUser)
+	self.engine.PUT("/api/v1/users/:id", self.updateUser)
+
+}
+
+// get information about all users. Pagination implemented.
+func (h *httpHandler) getUsersPagination(c echo.Context) error {
+	user := []domian.ApiResponse{}
+
+	page, err := strconv.Atoi(c.QueryParam("page")) // get page number
+	if err != nil {
+		log.Errorf("getUsersPagination. cannot get page number. error - %s", err)
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+	}
+
+	limit, err := strconv.Atoi(c.QueryParam("limit"))
+	if err != nil {
+		log.Errorf("getUsersPagination. cannot get limit value. error - %s", err)
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+	}
+
+	totalPageQty, err := h.service.GetPageQty(limit)
+	if err != nil {
+		log.Errorf("getUsersPagination. cannot get page q-ty. error - %s", err)
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+	}
+
+	if page > totalPageQty || page <= 0 {
+		log.Errorf("getUsersPagination. incorrect page number.")
+		return echo.NewHTTPError(http.StatusInternalServerError, "page number out of the available page range")
+	}
+
+	user, err = h.service.GetUsersPage(page, limit)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+	}
+
+	return c.JSON(http.StatusOK, user)
+}
+
+// get user by id. Basic Authentication implemented
+func (h *httpHandler) getUser(c echo.Context) error {
+	user := domian.ApiResponse{}
```

Comment:
the same

Reply:
solved with commit cb9c63563881cecdab598e4715ca0f33420b1c56